csrc/host_ir/executor.cpp

```diff
@@ -139,12 +179,13 @@ KernelArgumentHolder HostIrExecutor::run(
         "Output tensor not found in fusion outputs");
     auto out_tensor = output_args[out_idx].as<at::Tensor>();
 
-    c10::intrusive_ptr<c10d::Work> work = postSingleCommunication(
+    c10::intrusive_ptr<c10d::Work> work = validateAndPostSingleCommunication(
```

Comment:
I'll skip validation for HostIrExecutor. It's done elsewhere already. 

Input: https://github.com/NVIDIA/Fuser/blob/8a1028f84369f7709428a0e6aa6b2ebcbcdbd75b/csrc/tensor_metadata.cpp#L364

Output: https://github.com/NVIDIA/Fuser/blob/8a1028f84369f7709428a0e6aa6b2ebcbcdbd75b/csrc/runtime/allocations.cpp#L692

Reply:
Outputs are validated when they are not provided, i.e. output_args is empty. If not, then there is no validation. Inputs can be skipped, like you said.

I am looking at the callgraph to see what is the case, where output_args come pre-allocated.

Reply:
> what is the case, where output_args come pre-allocated

I looked for this before. I found only unit tests that explicitly call KernelExecutor, not via FusionExecutorCache.

Reply:
Right. FusionExecutorCache provides an empty output_args (https://github.com/NVIDIA/Fuser/blob/802f0427f163c9552a18e152d103402c93a775ac/csrc/runtime/fusion_kernel_runtime.cpp#L714).
I'll leave the validation in for just the `output` tensor and skip for inputs.